Makefile

```diff
@@ -2,23 +2,15 @@ CC=gcc
 CXX=g++
 NVCC=nvcc
 
-# We'll sacrifice longer compile times for broader compatibility out of the box.
 # Developer-users are suggested to change this in their make.inc, see:
 #   http://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/
-NVARCH = -arch=sm_70 \
-	-gencode=arch=compute_50,code=sm_50 \
-	-gencode=arch=compute_52,code=sm_52 \
-	-gencode=arch=compute_60,code=sm_60 \
-	-gencode=arch=compute_61,code=sm_61 \
-	-gencode=arch=compute_70,code=sm_70 \
-	-gencode=arch=compute_75,code=sm_75 \
-	-gencode=arch=compute_75,code=compute_75 
-
-CXXFLAGS= -DNEED_EXTERN_C -fPIC -O3 -funroll-loops -march=native -g -std=c++11
-#NVCCFLAGS=-DINFO -DDEBUG -DRESULT -DTIME
+#NVARCH = -arch=sm_70
```

Comment:
Maximum compatibility and compile times.  I found when I tested on other older gpu here that the original setup still did not always work as I thought (it would actually crash on SM35 cards, and give junk on SM50 for ex, which are still quite capable of running this).

I think if you are into performance tuning the last % out of something then it is worth exactly matching this to your hardware.  Otherwise probably not.  The only negative of not setting is the very first time you launch something, nv may try to compile some things on the fly, which you may notice as a lag on the first call, but this doesn't affect subsequent calls.  I left it with the note as a hint for people that want to `make.inc`.

Reply:
I went ahead and returned the multi compute/code arch line.  The crashes I observered were only if I forced arch without compiling all the other compute levels. My bad. Sorry for any confusion.

Reply:
Honestly, I think removing it was better. When I was testing this on different setups on FI it was causing crashes as well. For one, compiling on CUDA<10 will fail because nvcc won't recognize those (future) architectures. If we don't lose much performance, I think it's better to make it compile broadly and hint at how the user could optimize it if need be.

Reply:
Yes, this has been troubling both ways ... I emailed our acquaintance at Nvidia (he was the one answering these questions on SO anyway) about this.

As it stands right now, I think we would in fact want the PyPI wheel to have the gencode lines (ugh), so one solution is:
1) Remove from main Makefile
2) Create make.inc.cudaXX_fatbin (or similar) to be used for the wheel gen that has all gencode flags for cudaXX.

That should build and run for users (who may or may not want to optimize further), and allow me to control the wheel for maximum compat.

I do find that somewhat unsatisfying and want to wait to hear back.. I expect other people have written some utility code for this; maybe something we can use quickly, otherwise I'll implement as above.

Reply:
> I think we would in fact want the PyPI wheel to have the gencode lines 

Out of curiosity, why is that? At any rate, that solution sounds workable, albeit a bit ugly. I suggest we leave this for now and move it to an issue so that we can get the PR merged and move on to the next one (two?).

Reply:
We want nvcc to make code for all the GPUs it might encounter.  If I don't tell it to make code for sm35, at runtime will crash on my desktop (when using an older, but still capable GPU). I suspect that might be true for other compute capability levels like 52 that are still pretty common.

Sure, I'll make an issue. Can def dress it up in the other PR. Good idea.

Reply:
(This is confusing, because "compatibility" is ambiguous.  There is compatibility for the toolkit, the driver, the libraries, the GPU etc, and they are all important, and some are important at different times...)

Reply:
#37